docs/source/cuda/random.rst

```diff
@@ -64,3 +64,27 @@ Here is a sample program that uses the random number generator::
 
     compute_pi[blocks, threads_per_block](rng_states, 10000, out)
     print('pi:', out.mean())
+
+An example managing RNG state size and using a 3D grid
+''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+The size of the RNG states scales with the number of threads using the RNG, so
```

Comment:
I'm not sure about this. Should it be `RNG state`, the RNG holds a state-slot per thread but the state-slots themselves don't scale with thread count, they are a property of the RNG itself?

Reply:
I'm finding it hard to understand the question (in particular "the state-slots themselves don't scale with thread count, they are a property of the RNG itself?") , but I'll try to answer: I think of it as "RNG states" because as you describe, the RNG holds one state slot per thread.

Perhaps this would be better as:

```suggestion
The number of RNG states scales with the number of threads using the RNG, so
```

Reply:
I think the suggestion is clearer and nods towards the direct proportionality with thread count.

RE: "the state-slots themselves don't scale with thread count, they are a property of the RNG itself?"

The there's one slot per thread and each slot contains a single RNG state. The size of this slot is the size of a single RNG state and is a fixed property of the RNG in use, e.g. Mersenne Twister has a different size to some `xoroshiro` variant? I think we're probably talking about the same things but with varying degrees of overloading on "state" and "RNG".